src/basic/user-util.c

```diff
@@ -709,6 +709,30 @@ int take_etc_passwd_lock(const char *root) {
         return fd;
 }
 
+int take_etc_passwd_lock_at(int dir_fd) {
+
+        struct flock flock = {
+                .l_type = F_WRLCK,
+                .l_whence = SEEK_SET,
+                .l_start = 0,
+                .l_len = 0,
+        };
+
+        int fd, r;
+
+        fd = openat(dir_fd, ETC_PASSWD_LOCK_PATH, O_WRONLY|O_CREAT|O_CLOEXEC|O_NOCTTY|O_NOFOLLOW, 0600);
```

Comment:
this needs to use chase_symlinks.

i mean ETC_PASSWD_LOCK_PATH is an absolute path, hence dir_fd is actually entirely ignore